cppapi/server/fwdattrdesc.cpp

```diff
@@ -180,7 +180,10 @@ bool FwdAttr::validate_fwd_att(vector<AttrProperty> &prop_list,const string &dev
 	}
 	catch (...) {}
 
-	if (root_att_db_defined == true)
+	//check if full_root_att is already set
+	if (full_root_att.size()!=0 and full_root_att.compare(RootAttNotDef)!=0)
```

Comment:
Actually it looks like this is part of the C++ standard, but very rarely used:
http://stackoverflow.com/questions/4251699/is-it-okay-to-use-and-or-etc-instead-of

I can change it by && if you prefer.